### PR TITLE
FISH-961 Document MicroProfile 4.0 upgrade

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

File: docs/modules/ROOT/pages/documentation/microprofile/README.adoc [hunk 1]
```diff
@@ -64,11 +68,37 @@ supported by Payara Server and Payara Micro, an application targeting version
...
+* https://github.com/eclipse/microprofile-opentracing/releases/tag/2.0[OpenTracing 2.0]
+* https://github.com/eclipse/microprofile-open-api/releases/tag/2.0[OpenAPI 2.0]
+* https://github.com/eclipse/microprofile-rest-client/releases/tag/2.0[Rest Client 2.0]
+
+| [source, xml]
+----
+<dependency>
+    <groupId>org.eclipse.microprofile</groupId>
+    <artifactId>microprofile</artifactId>
+    <version>4.0</version>
```
Comment (rdebusscher) [inline docs/modules/ROOT/pages/documentation/microprofile/README.adoc:92]: Version 4.0 is broken, users need to use `4.0.1`

**Review by rdebusscher — CHANGES_REQUESTED** [review]

**Review by fturizo — APPROVED** [review]
Aside from @rdebusscher's comment, everything looks good

**Review by MattGill98 — CHANGES_REQUESTED** [review]
Good so far, and we link to the spec docs so not every little change needs documenting. There are, however, a few implementation details that need documenting.

Firstly, each individual spec page lists a version that needs changing. Secondly, off the top of my head:

__Health__
`MP_HEALTH_BACKWARD_COMPATIBILITY_ENABLED` doesn't work anymore I don't think.
`@Health` annotation isn't present anymore.

__OpenAPI__
New property `mp.openapi.schema.`

__Metrics__
Now contains advice on handling multiple apps.
Base and Vendor metrics are now "per-application"

__FT__
Changed back to using ManagedExecutors. I believe this was changed and then reverted, so the state of the docs may be correct on this one.

I'd strongly suggest reading the PR for each of these implementations (with the addition of Patrik's second FT PR) for details to make sure our docs are still valid.

**Comment by fturizo** [comment]
@MattGill98, apologies for the late reply. Technically the specifics you mentioned should have been documented separately by the implementor of the changes of each spec, so keep this in mind for the future. Since we only document the main differences and proprietary functionality from the spec documents, here are my 2 cents on the changes you mentioned:

* Health:  We do mention the use of the config property and the annotation so indeed we need to update this
* OpenAPI: The property needs to be added to the table in the corresponding section
* Metrics: I reviewed the spec changes and didn't see any changes that we need to document
* FT: This one is confusing. Our docs establish that we allow the configurable executor in one place and then we later state the contrary, so this one needs to be fixed. @pdudits, can you clarify which one is it? 

@MeroRai, can you make the other changes part of this PR as well?

**Comment by MeroRai** [comment]
@fturizo, I have document changes made in all of the MicroProfile specs. In regards, to the Fault Tolerance, it is broken (read the documentation for more info). I have discussed this with @MattGill98 and it will be fixed in the future release as it is too late to be fixed in the upcoming release.

File: docs/modules/ROOT/pages/documentation/microprofile/faulttolerance.adoc [hunk 2]
```diff
@@ -79,15 +79,15 @@ image:microprofile/fault-tolerance.png[Set Fault Tolerance Configuration]
 [[using-asadmin-commands]]
 === Using Asadmin Commands
 
-NOTE: Since Payara Platorm 5.2020.2, the configuration no longer allows configuring user defined executor to run the fault tolerance execution. Instead, Fault Tolerance uses two dedicated executors that are well suited for the usage pattern typically experienced when executing Fault Tolerance semantics. It's possible to configure the maximum size of these pools to scale them up or down.
+NOTE: On Payara Platform 5.2021.1, it is a known issue that the `set-fault-tolerance-configuration` command is broken and it can't be used to configure user-defined executor to run the fault tolerance execution. To work around this issue, use the following `set` command to configure managed executor pool: `asadmin set configs.config.server-config.microprofile-fault-tolerance-configuration.managed-executor-service` and similar use the following  command to configure scheduled managed executor pool: `asadmin set configs.config.server-config.microprofile-fault-tolerance-configuration..managed-scheduled-executor-service`. Finally, a server restart is required for the commands to take effect.
```
Comment (rdebusscher) [inline docs/modules/ROOT/pages/documentation/microprofile/faulttolerance.adoc:82]: Should we mention here already that a default executor is used an no longer a dedicated executor?

File: docs/modules/ROOT/pages/documentation/microprofile/healthcheck.adoc [hunk 3]
```diff
@@ -34,8 +34,8 @@ The MicroProfile Healthchecks API defines the following list of REST endpoints t
 |On invocation, all registered health checks annotated with `@Readiness` are performed and the results aggregated.
 
 |`/health`
-|Readiness + Liveness + Backward compatible
-|On invocation, all registered health checks annotated with `@Liveness`, `@Readiness` and `@Health` are performed and the results aggregated.
+|Readiness + Liveness 
+|On invocation, all registered health checks annotated with `@Liveness`and `@Readiness` are performed and the results aggregated.
```
Comment (rdebusscher) [inline docs/modules/ROOT/pages/documentation/microprofile/healthcheck.adoc:38]: missing space ? 

`@Liveness`and

**Review by rdebusscher — APPROVED** [review]
Some comments but can be merged as is.

**Comment by fturizo** [comment]
Looks good, @MeroRai. I did a couple of grammar and layout fixes prior to the merge.

**Pull request merged**